### ISSUE TITLE
Is there a way of analyzing the context on the Completion Suggester?

### ISSUE THREAD

**Issue by AIsaac08** [elastic/elasticsearch#16064]
We know this is a great feature for the completion suggester, but as far as I know is too serious with the argument it takes. My question is if it has any way of analyzing the context so cases like the examples below cool match.

*Example 1:*
```JSON
GET my_index/_suggest
{
  "Autocomplete": {
    "text": "T",
    "completion": {
      "field": "autocomplete_field",
      "size": 15,
      "context": {
        "color": "red"
      }
    }
  }
}
```

*Example 2:*
```JSON
GET my_index/_suggest
{
  "Autocomplete": {
    "text": "T",
    "completion": {
      "field": "autocomplete_field",
      "size": 15,
      "context": {
        "color": "Red"
      }
    }
  }
}
```
Thanks

**Comment by clintongormley** [comment]
Hi @AIsaac08 

No there isn't.  Contexts are intended as simple filters - the kind of thing you'd select from a prepopulated list on a website.

**Comment by AIsaac08** [comment]
Ok, thanks. By the way, I'm facing issues when some elements in the `context` contains accents, of course I'm querying with the accent, but they fail to match however.

**Comment by clintongormley** [comment]
Hi @AIsaac08 

I've just tried out a context like `réd` and confirm that it is not working in 2.1 or before.  We're not going to fix this as, in ES 2.2.0, we have [completely rewritten](https://www.elastic.co/guide/en/elasticsearch/reference/2.x/search-suggesters-completion.html) the completion suggester. I can confirm that contexts with accents work correctly there.

**Comment by AIsaac08** [comment]
Thanks! When will ES 2.2 be available as a release?
